File: api/src/monster/types/size.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, sqlx::Type, Serialize, Deserialize)]
#[sqlx(type_name = "size", rename_all = "lowercase")]
#[serde(rename_all = "snake_case")]
pub enum Size {
    Tiny,
    Small,
    Medium,
    Large,
    Huge,
    Gargantuan,
}

File: view/src/lib/types/Size.ts
export enum Size {
  Tiny = "tiny",
  Small = "small",
  Medium = "medium",
  Large = "large",
  Huge = "huge",
  Gargantuan = "gargantuan",
}

export const DisplaySize: Readonly<Record<Size, string>> = {
  [Size.Tiny]: "Tiny",
  [Size.Small]: "Small",
  [Size.Medium]: "Medium",
  [Size.Large]: "Large",
  [Size.Huge]: "Huge",
  [Size.Gargantuan]: "Gargantuan",
} as const;

export const Sizes: Size[] = Object.values(Size);
